atexit.go

```diff
@@ -50,6 +52,8 @@ func runHandler(handler func()) {
 }
 
 func runHandlers() {
+	handlersLock.RLock()
```

Comment:
I prefer to use [sync.Once](https://golang.org/pkg/sync/#Once) to make sure we don't run handlers twice.

Reply:
You also need to ensure you do not run handlers at the same time as other goroutine adds another handler.

Reply:
Agree